### PR TITLE
Delete the public TriggerDiagnosticDescriptor type.

### LINKED ISSUE #789: TriggerDiagnosticDescriptor doesn't prevent empty preview
As discussed in PR #773 for Issue #466, using a TriggerDiagnosticDescriptors does not prevent the preview pane from showing.

### PR DESCRIPTION
**Pull request by shyamnamboodiripad**
And seal its base type DiagnosticDescriptor.

(Fixes #789)

### CODE REVIEW

**Comment by shyamnamboodiripad** [comment]
@tmeschter @mavasani @heejaechang @srivatsn could you please take a look?

File: src/Features/Core/Diagnostics/Analyzers/RemoveUnnecessaryImportsDiagnosticAnalyzerBase.cs [hunk 1]
```diff
@@ -27,7 +26,11 @@ internal abstract class RemoveUnnecessaryImportsDiagnosticAnalyzerBase : Diagnos
                                                                                             isEnabledByDefault: true,
                                                                                             customTags: DiagnosticCustomTags.Unnecessary);
 
-        private static readonly DiagnosticDescriptor s_fixableIdDescriptor = new TriggerDiagnosticDescriptor(DiagnosticFixableId);
+        private static readonly DiagnosticDescriptor s_fixableIdDescriptor = new DiagnosticDescriptor(DiagnosticFixableId,
```
Comment (srivatsn) [inline src/Features/Core/Diagnostics/Analyzers/RemoveUnnecessaryImportsDiagnosticAnalyzerBase.cs]: Maybe add a comment about the reasons for message=null and the notconfigurable tag?

**Comment by srivatsn** [comment]
:+1:

File: src/Test/Utilities/DescriptorFactory.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Roslyn.Utilities;
+
+namespace Roslyn.Test.Utilities
+{
+    public static class DescriptorFactory
+    {
+        public static DiagnosticDescriptor CreateSimpleDescriptor(string id)
```
Comment (mavasani) [inline src/Test/Utilities/DescriptorFactory.cs]: Doc comments here would be helpful.

**Comment by mavasani** [comment]
:+1:

**Comment by heejaechang** [comment]
:+1:

File: src/Features/Core/Diagnostics/Analyzers/RemoveUnnecessaryImportsDiagnosticAnalyzerBase.cs (hunk 1 above)
Comment (shyamnamboodiripad) [inline src/Features/Core/Diagnostics/Analyzers/RemoveUnnecessaryImportsDiagnosticAnalyzerBase.cs]: Sure I'll add a comment for this.

File: src/Test/Utilities/DescriptorFactory.cs (hunk 2 above)
Comment (shyamnamboodiripad) [inline src/Test/Utilities/DescriptorFactory.cs]: I didn't add doc comments since this is for tests only :) But will add.

**Comment by tmeschter** [comment]
:+1:

**Pull request merged**